src/optimizer/query_planner.c

```diff
@@ -3733,18 +3585,15 @@ qo_plan_cmp (QO_PLAN * a, QO_PLAN * b)
 	  return PLAN_COMP_GT;
 	}
 
-      if (!qo_is_index_iss_scan (a) && !qo_is_index_loose_scan (a))
+      if (a->plan_un.scan.index && a->plan_un.scan.index->head->groupby_skip)
```

Comment:
Is it okay not to check `qo_is_iscan ()` ?
Even in AS-IS, it is not restricted to be executed only when `qo_is_iscan ()` is true.

However, since `(QO_PLAN *)->plan_un` is a union,
I think there may be garbage values in `(QO_PLAN *)->plan_un.scan.index` even though it is not an index scan.
Would not it be dangerous to move to `(QO_PLAN *)->plan_un.scan.index` when garbage values are stored?

Reply:
I thought again that there is no problem because `a->plan_type == QO_PLANTYPE_SCAN` is checked first.